PhysicsTools/NanoAOD/plugins/DispJetTableProducer.cc

```diff
@@ -0,0 +1,503 @@
+// -*- C++ -*-
+//
+// Package:    PhysicsTools/EXOnanoAOD
+// Class:      DispJetTableProducer
+//
+/**\class DispJetTableProducer
+
+ Description: Additional variables for displaced vertices involving at least one lepton and other tracks
+
+*/
+//
+// Original Author:  Kirill Skovpen
+//         Created:  Sat, 1 Mar 2025 08:37:01 GMT
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/PatCandidates/interface/Electron.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "DataFormats/MuonReco/interface/MuonSelectors.h"
+
+// nanoAOD include files
+#include "DataFormats/NanoAOD/interface/FlatTable.h"
+
+// object specific include files
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+#include "TrackingTools/IPTools/interface/IPTools.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+
+class DispJetTableProducer : public edm::stream::EDProducer<> {
+protected:
+  edm::ESGetToken<TransientTrackBuilder, TransientTrackRecord> ttbToken_;
+  edm::EDGetTokenT<std::vector<pat::Electron>> electronTag_;
+  edm::EDGetTokenT<std::vector<pat::Muon>> muonTag_;
+  edm::EDGetTokenT<reco::VertexCollection> vtxTag_;
+  edm::EDGetTokenT<reco::VertexCollection> secVtxTag_;
+
+  const TransientTrackBuilder* theB;
```

Comment:
`theB` seems to be used only inside the `produce()` function, so this data member does not seem to be needed.